examples/advanced_diffusion_training/train_dreambooth_lora_sdxl_advanced.py

```diff
@@ -1539,11 +1594,13 @@ def load_model_hook(models, input_dir):
     # pooled text embeddings
     # time ids
 
-    def compute_time_ids():
+    def compute_time_ids(original_size=None, crops_coords_top_left=None):
         # Adapted from pipeline.StableDiffusionXLPipeline._get_add_time_ids
-        original_size = (args.resolution, args.resolution)
+        if original_size is None:
+            original_size = (args.resolution, args.resolution)
         target_size = (args.resolution, args.resolution)
-        crops_coords_top_left = (args.crops_coords_top_left_h, args.crops_coords_top_left_w)
+        if crops_coords_top_left is None:
+            crops_coords_top_left = (args.crops_coords_top_left_h, args.crops_coords_top_left_w)
```

Comment:
I think we should really remove `--crops_coords_top_left_h` from the CLI args because then we are always explicitly conditioning on a crop geometry that is down right incorrect. WDYT?

Reply:
yes!

Reply:
Thinking about it now, we can also remove:
https://github.com/huggingface/diffusers/blob/1922313e48cb129d18b28d07a5ba690ea6e17ec7/examples/advanced_diffusion_training/train_dreambooth_lora_sdxl_advanced.py#L1616

and:
https://github.com/huggingface/diffusers/blob/1922313e48cb129d18b28d07a5ba690ea6e17ec7/examples/advanced_diffusion_training/train_dreambooth_lora_sdxl_advanced.py#L1643
https://github.com/huggingface/diffusers/blob/1922313e48cb129d18b28d07a5ba690ea6e17ec7/examples/advanced_diffusion_training/train_dreambooth_lora_sdxl_advanced.py#L1645

right? 🤔

Reply:
Not sure about:

```
add_time_ids = torch.cat([add_time_ids, class_time_ids], dim=0)
```

as I don't know what `class_time_ids` is. But, yes to the others.

Reply:
Yes! As long as the time ids are properly computed when prior-preservation is enabled, I think we should be fine. 

I also think it's totally okay to compute the time ids like we're during the training loop as it's almost no computation. But it greatly improves the readability overall.

Reply:
Actually, now I'm not sure about `--crops_coords_top_left_h`. 
I agree that it'll be incorrect, but if the images are not cropped, then it'd be right no? maybe it's not a likely use case, but if a user has 1024x images that are off center isn't it correct/useful then?

Reply:
https://github.com/huggingface/diffusers/blob/03373de0db15512f4e785919409142f94ef86dcb/examples/text_to_image/train_text_to_image_lora_sdxl.py#L851 

provides a nice reference

Reply:
Thanks! yeah this is the reference I used to add the micro conditioning in here as well. So I totally agree about the calculation being incorrect, but thought maybe in some cases, if we make changes to the current preprocessing that is, it could make sense and lead to better results? (when users know in advance maybe that their data is mostly off-center etc)

Reply:
but I actually think now it might make it overly complicated and unnecessary

Reply:
That should be a bit on the user side I think. We could definitely point out these concerns centrally here: https://github.com/huggingface/diffusers/issues/6397. But in the official examples, I think it's better to have a harmonized approach.